front/types/chat.ts

```diff
@@ -14,9 +14,7 @@ export type ChatRetrievedDocumentType = {
 
 export type ChatMessageType = {
   role: "user" | "retrieval" | "assistant" | "error";
-  runRetrieval?: boolean;
-  runAssistant?: boolean;
-  message?: string; // for `user`, `assistant` and `error` messages
+  message?: string; // for `user`, `assistant` and `error` messages, and `retrieval` messages storing a query
```

Comment:
Let’s type the query explicitly 👍

Reply:
ok DB migration then?

Reply:
also are there other places where I should have typed in your opinion? generally and specifically
(it's been 10 years since I used a typed language :sweat_smile: so I need to get re-used to it)

Reply:
Type as much as you can basically :)

```
query?: { xxx :string, yyy: number }
```

I guess it's fine to not persist this one in DB for now, but if needed then probably store it as a json blob